addons/website/static/src/snippets/s_embed_code/options.js

```diff
@@ -0,0 +1,86 @@
+odoo.define('website.s_embed_code_options', function (require) {
+'use strict';
+
+const Dialog = require('web.Dialog');
+const core = require('web.core');
+const options = require('web_editor.snippets.options');
+
+const _t = core._t;
+
+options.registry.EmbedCode = options.Class.extend({
+    events: {
+        'click .o_we_edit_code': '_onEditClick',
+    },
+
+    _onEditClick: async function () {
+        const $container = this.$target.find('.o_embedded');
+        let code = $container.html().trim();
+
+        const libsProm = this._loadLibs({
+            jsLibs: [
+                '/web/static/lib/ace/ace.js',
+                '/web/static/lib/ace/mode-xml.js',
+                '/web/static/lib/ace/mode-qweb.js',
+            ],
+        });
+
+        await Promise.all([libsProm]);
+
+        await new Promise(resolve => {
+            const $content = $(core.qweb.render('website.custom_code_dialog_content'));
+            const aceEditor = this._renderAceEditor($content.find('.o_ace_editor_container')[0], code || '');
+            const dialog = new Dialog(this, {
+                title: _t("Edit embedded code"),
+                $content,
+                buttons: [
+                    {
+                        text: _t("Save"),
+                        classes: 'btn-primary',
+                        click: async () => {
+                            const $html = $('<div>' + aceEditor.getValue() + '</div>');
```

Comment:
Is there a point in wrapping in a div first?

Reply:
The `<div>` is put there because jQuery's `$(html)` might inadequately detect the parameter as being a selector expression.
E.g.: giving the following
```html
Some text at the beginning
<div>Some element</div>
Some text at the end
```
as parameter leads to a `Syntax error, unrecognized expression`.

Reply:
Ok, then my question is why parsing the `aceEditor.getValue()` at all? Just one line `containerEl.innerHTML = newHTML;` should be enough :wink: